hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -76,6 +80,10 @@
     protected final InterceptorRegistry interceptorRegistry = new InterceptorRegistry();
     protected final IFunction<Object, Data> toDataFunction = new ObjectToData();
     protected final ConstructorFunction<Void, RecordFactory> recordFactoryConstructor;
+    protected final boolean globalIndex;
+    protected final Indexes indexes;
+    protected final ConcurrentHashMap<String, Boolean> indexDefinitions = new ConcurrentHashMap<String, Boolean>();
```

Comment:
can you please put some short javaDoc to explain `indexDefinitions` and its usage?